src/main/content/_assets/js/javadoc-redirect.js

```diff
@@ -1,68 +1,94 @@
 $(function () {
-    if (
-        window.top.location.href.includes('/docs/modules/reference/microprofile-')
-    ) {
-        var jd = window.top.location.href;
-        var version = jd.substring(
-            jd.indexOf('microprofile-') + 13,
-            jd.indexOf('microprofile-') + 16
-        );
-        jd = jd.substring(jd.indexOf('microprofile-') + 25);
-        var pack = jd.substring(0, jd.lastIndexOf('/'));
-        var port =
-      window.top.location.port !== '' ? ':' + window.top.location.port : '';
-        window.top.location.href =
+  if (
+      window.top.location.href.includes('/docs/modules/reference/microprofile-')
+  ) {
+      var jd = window.top.location.href;
+      var version = jd.substring(
+          jd.indexOf('microprofile-') + 13,
+          jd.indexOf('microprofile-') + 16
+      );
+      jd = jd.substring(jd.indexOf('microprofile-') + 25);
+      var pack = jd.substring(0, jd.lastIndexOf('/'));
+      var port =
+    window.top.location.port !== '' ? ':' + window.top.location.port : '';
+    if(pack) {
+      window.top.location.href =
       'https://' +
       window.top.location.hostname +
       port +
       '/docs/latest/reference/javadoc/microprofile-' +
       version +
-      '-javadoc.html#package=' +
+      '-javadoc.html?package=' +
       pack +
       '/package-frame.html&class=' +
       jd;
```

Comment:
we can also further simplify the code by putting the content inside if (pack) code in a method ifPackageExists(port, version, pack, jd)

Reply:
on next PR I will make that refactoring